src/pudl/io_managers.py

```diff
@@ -193,7 +193,7 @@ def _get_fk_list(self, table: str) -> pd.DataFrame:
         method collapses foreign keys with multiple fields into one record for
         readability.
         """
-        with self.engine.connect() as con:
+        with self.engine.begin() as con:
```

Comment:
Incredibly, now that all of our interactions with the database are centralized in the `IOManager` classes, this is the only place we need to change... anything to work with SQLAlchemy 2.0!